### PR TITLE
Additional legend exports

### PR DESCRIPTION
**Pull request by robertjanetzko**
i've added the following to exportlegends
* export poetic, musical and dance froms
* export written content, contents of artifacts
* additional site & structure date
* hf sex and race
* entity positions and assignments

### CODE REVIEW

File: scripts/exportlegends.lua [hunk 1]
```diff
@@ -236,6 +236,72 @@ function export_more_legends_xml()
...
+    for wcK, wcV in ipairs(df.global.world.written_contents.all) do
+        file:write("\t<written_content>\n")
+        file:write("\t\t<id>"..wcV.id.."</id>\n")
+        file:write("\t\t<title>"..wcV.title.."</title>\n")
+        file:write("\t\t<page_start>"..wcV.page_start.."</page_start>\n")
+        file:write("\t\t<page_end>"..wcV.page_end.."</page_end>\n")
+        for refK, refV in pairs(wcV.refs) do
+            file:write("\t\t<reference>\n")
+            file:write("\t\t\t<type>"..df.general_ref_type[refV:getType()].."</type>\n")
+            if refV:getType() == 64     then file:write("\t\t\t<id>"..refV.anon_1.."</id>\n") -- written content
```
Comment (lethosor) [inline scripts/exportlegends.lua]: This section should use checks like `if refV:getType() == df.general_ref_type.WRITTEN_CONTENT` instead. You can get the appropriate names in the `lua` interpreter with `~df.general_ref_type[64]`, etc. (or use `@df.general_ref_type` for a complete list).

**Comment by lethosor** [comment]
Travis was complaining because of f02a504a0087be23258b7ccf021b6991d510fae2 (should be fixed in 15a9d7f now) - sorry about that.
I haven't had much time to look at things recently, but I should be able to this weekend. Is there anything else you want to add?

**Comment by robertjanetzko** [comment]
i wanted to add the information for occasions, performances and ceremonies, but i couldn't find them in the structures. i guess they hide somewhere in 'historical_entity' in this block (probably in the pointer)

```
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <stl-vector type-name='int32_t' comment='v0.42.01'/>
        <pointer comment='v0.42.01'/>
```

**Comment by lethosor** [comment]
Yeah, those probably haven't been mapped yet. It could be that some of those structures are in the vectors, too - 32-bit integers and pointers are the same size, so they can be confused sometimes.

**Pull request merged**